src/main/java/org/jenkinsci/plugins/workflow/multibranch/JobPropertyStep.java

```diff
@@ -113,7 +120,6 @@
             try {
                 for (JobProperty prop : job.getAllProperties()) {
                     if (prop instanceof BranchJobProperty) {
-                        // TODO do we need to define an API for other properties which should not be removed?
```

Comment:
If you are deciding that there might be corner cases not fully covered by the main logic, best to leave a comment explaining that `BranchJobProperty` is considered especially critical, etc.